cmd/thanos/sidecar.go

```diff
@@ -262,43 +263,52 @@ func runSidecar(
 
 		exemplarSrv := exemplars.NewPrometheus(conf.prometheus.url, c, m.Labels)
 
-		infoSrv := info.NewInfoServer(
-			component.Sidecar.String(),
-			info.WithLabelSetFunc(func() []labelpb.ZLabelSet {
+		var infoOptions []info.ServerOptionFunc
+		if !promAgentModeEnabled {
+			infoOptions = append(infoOptions, info.WithLabelSetFunc(func() []labelpb.ZLabelSet {
 				return promStore.LabelSet()
 			}),
```

Comment:
Let's make this a little easier to read,
```suggestion
			infoOptions = append(infoOptions,
				info.WithLabelSetFunc(func() []labelpb.ZLabelSet {
					return promStore.LabelSet()
				}),
```